datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -100,6 +104,7 @@ impl ParquetExec {
         base_config: FileScanConfig,
         predicate: Option<Expr>,
         metadata_size_hint: Option<usize>,
+        parquet_file_reader_factory: Option<Arc<dyn ParquetFileReaderFactory>>,
```

Comment:
I think a doc comment explaining what this is would be good.

Also this is technically a breaking change

Reply:
we could take a different route and create builder like method that allows to override that, what do you think ? 

`with_ parquet_file_reader_factory(mut self, ...) -> Self`

Reply:
Sounds good to me :+1: